xlators/cluster/afr/src/afr-inode-read.c

```diff
@@ -912,16 +912,12 @@ afr_update_local_dicts(call_frame_t *frame, dict_t *dict, dict_t *xdata)
         UNLOCK(&frame->lock);
     }
 
-    if (dict != NULL) {
-        if (dict_copy(dict, local->dict) == NULL) {
-            goto done;
-        }
```

Comment:
Will do the required changes. In that case the above check for (dict!= NULL ) and (xdata!=NULL) shouldn't be removed as well as due to the same reasons.

Reply:
dict_copy() internally checks for dict null.

Reply:
`dict_copy()` checks for NULL but writes a warning message to the log. A `NULL` dict when calling that function is considered an error, so checking it before calling is the right thing to do.